.github/workflows/unmanaged_dependency_check.yaml

```diff
@@ -0,0 +1,26 @@
+on:
+  pull_request:
+name: Unmanaged dependency check
+jobs:
+  unmanaged_dependency_check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-java@v3
+      with:
+        distribution: temurin
+        java-version: 11
+    - name: Install modules
+      shell: bash
+      run: |
+        mvn -B -V -ntp -DskipTests=true \
+        -Dclirr.skip=true \
+        -Denforcer.skip=true \
+        -Dmaven.javadoc.skip=true \
+        -Dgcloud.download.skip=true \
+        -T 1C \
+        clean install
+    - name: Unmanaged dependency check
+      uses: googleapis/sdk-platform-java/java-shared-dependencies/unmanaged-dependency-check@unmanaged-dependencies-check-latest
```

Comment:
By using tag `unmanaged-dependencies-check-latest`, the check always runs against the latest released version of java shared dependencies.

No need to update the tag when the java shared dependencies is updated.

Reply:
this looks good - is the plan to maintain this in googleapis/synthtool or in here?

Reply:
First we'll gradually implement repositories whose owner reviewed [the idea](https://docs.google.com/document/d/1D6aeJr46PbxCSBhPCBqaSybhVNlFNbfVynVkRhF6f84/edit?usp=sharing). Then we'll try to make it to the template.

Reply:
Great, thanks! Is there an issue for tracking purposes?

Reply:
Hi @kolea2, what type of information do you want to track in the issue?

Reply:
a simple tracking issue for this work is sufficient. Thanks!

Reply:
We just created b/320677260.